### ISSUE TITLE
[Bug]: In browse mode, screen reader not announcing the selected/not selected state for the legends.

### ISSUE THREAD

**Issue by gilenzi** [microsoft/fluentui#31597]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 13th Gen Intel(R) Core(TM) i5-1335U
    Memory: 3.94 GB / 15.69 GB
  Browsers:
    Edge: Chromium (125.0.2535.79)
    Internet Explorer: 11.0.22621.3527
```

### Are you reporting Accessibility issue?

![image](https://github.com/microsoft/fluentui/assets/78443187/2a086bab-5dae-4ecc-bd19-51784aaceec4)
Screen reader in browse mode not announcing the selected/not selected state for the legends.

### Reproduction

https://scc-storybook.azurewebsites.net/storybook-vnext/iframe.html?args=&globals=&id=components-card-content-dualreportcardcontent--default&viewMode=story

## Actual Behavior
In browse mode, screen reader is not announcing the state of the legends.

## Expected Behavior
Screen reader should announce the state of the legends in browse mode.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by behowell** [comment]
Hi @gilenzi, please provide a minimal repro that includes source code (e.g. using codepen or stackblitz).

**Comment by gilenzi** [comment]
I added image, because that is screen reader issue related for "Legends" component. So I don't know which code to include. Here are repro steps: 
1. Open URL in the Edge browser.
2.  Storybook M365 SCC Components page opens.  
3.  Navigate to left panel and activate 'Card Content' under components.
4. Activate 'DualreportcardContent' tab > Default.
5. Activate 'Open canvas in in tab'. 
7. The new page will be opened in the new tab.
8. Now, navigate to the graph legends in browse mode and observe the issue.

**Comment by behowell** [comment]
As far as I can tell, it is setting `aria-selected="true"` correctly when a single legend item is selected. And Narrator does read "selected" for those items.

I noticed that when there is no selection, all legend items appear selected, but they all have `aria-selected="false"`. Is that the case you're referring to?

<img width="438" alt="image" src="https://github.com/microsoft/fluentui/assets/48106640/99cfc3ad-2107-4802-8978-1bc63e11b2f7">

I'm not an expert on charting, so I've assigned this to the charting team to take a further look.

**Comment by AtishayMsft** [comment]
@gilenzi Could you share a video of the issue.
I am seeing that selected and not selected state is announcing on selection/deselection of legends.
And the not selected legend states are announced on keyboard navigation.

**Comment by smhigley** [comment]
I can't say for sure without more info on the screen reader used and the browser (guessing NVDA because of "browse mode"?), but this is almost certainly the specific NVDA/JAWS behavior of not announcing "selected" when you navigate to selected listbox options. This stems from historical Windows desktop choices around selected state following focus, and is not a bug.

You should hear the selected state when you toggle selection, and you should hear "not selected" announced when you move onto options that are not selected (assuming this is NVDA; this behavior varies by screen reader, since it is a screen reader UX choice).
